mdata/store_cassandra.go

```diff
@@ -584,9 +588,14 @@ LOOP:
 		}
 		err := outcome.i.Close()
 		if err != nil {
+			if err == context.Canceled || err == context.DeadlineExceeded {
+				// query was aborted.
+				return nil, nil
+			}
 			tracing.Failure(span)
 			tracing.Error(span, err)
 			errmetrics.Inc(err)
```

Comment:
this line triggers this code:
```
func (m *ErrMetrics) Inc(err error) {
	if err == gocql.ErrTimeoutNoResponse {
		m.cassErrTimeout.Inc()
	} else if err == gocql.ErrTooManyTimeouts {
		m.cassErrTooManyTimeouts.Inc()
	} else if err == gocql.ErrConnectionClosed {
		m.cassErrConnClosed.Inc()
	} else if err == gocql.ErrNoConnections {
		m.cassErrNoConns.Inc()
	} else if err == gocql.ErrUnavailable {
		m.cassErrUnavailable.Inc()
	} else if strings.HasPrefix(err.Error(), "Cannot achieve consistency level") {
		m.cassErrCannotAchieveConsistency.Inc()
	} else {
		m.cassErrOther.Inc()
	}
}
```
seems weird that various gocql timeouts result in err metrics being incremented, whereas context.DeadlineExceeded does not. I think we should also increment a timeout metric when context.DeadlineExceeded triggers

Reply:
store_cassandra.go does not set a context Timeout on the read path. 

So if context.Canceled or context.DeadlineExceeded is returned,  it is because the caller wanted us to give up on the request.  There was no issue preventing us from executing the request.

currently, we dont set a deadline on the request context anywhere, so context.DeadlineExceeded will never be returned.  It is just here for completeness incase we oneday add a deadline to the context.

Reply:
wouldn't we want context.DeadlineExceeded to mark the spans as failed and the ErrMetrics to be incremented? it seems to me any such deadline being hit is a failure that should be reported

Reply:
No.  If DeadlineExceeded is reached it is because the caller set a timeout and they should emit their own error metrics and set the span to failed.

The error would have nothing to do with cassandra, so you should not increment the cassandraErr counters

Reply:
fair enough